lib/generators/rspec/scaffold/scaffold_generator.rb

```diff
@@ -32,9 +32,7 @@ def generate_controller_spec
           controller_class_path,
           "#{controller_file_name}_controller_spec.rb"
         )
-        if options[:api]
-          template 'api_controller_spec.rb', template_file
-        else
+        unless options[:api]
```

Comment:
Basically you need to switch on your request vs controller format here, and still respect existing installtions.